packages/sanity/README.md

```diff
@@ -0,0 +1,214 @@
+
+### Recommendations
+
+you must make the dataset private otherwise the data of your users will be public for everyone
+
+```sh
+sanity dataset visibility set <datasetName> private
+```
+
+## Requirements
+
+- Sanity Token for Read+Write
+
+## Installation
+
+### yarn
+
+```sh
+yarn add next-auth-sanity
+```
+
+### npm
+
+```sh
+npm i next-auth-sanity
+```
+
+## Usage
+
+[Full Example](https://github.com/Fedeya/next-auth-sanity/tree/main/examples/full-example)
+
+```ts
+import NextAuth, { NextAuthOptions } from "next-auth"
+import Providers from "next-auth/providers"
+import { NextApiRequest, NextApiResponse } from "next"
+import { SanityAdapter, SanityCredentials } from "next-auth-sanity"
+import { client } from "your/sanity/client"
+
+const options: NextAuthOptions = {
+  providers: [
+    Providers.GitHub({
+      clientId: process.env.GITHUB_CLIENT_ID,
+      clientSecret: process.env.GITHUB_CLIENT_SECRET,
+    }),
+    SanityCredentials({ client }), // only if you use sign in with credentials
+  ],
+  session: {
+    jwt: true,
+  },
+  adapter: SanityAdapter({ client }),
+}
+
+export default (req: NextApiRequest, res: NextApiResponse) =>
+  NextAuth(req, res, options)
+```
+
+### Sanity Schemas
+
+```ts
+// user
+
+export default {
+  name: "user",
+  title: "User",
+  type: "document",
+  fields: [
+    {
+      name: "name",
+      title: "Name",
+      type: "string",
+    },
+    {
+      name: "email",
+      title: "Email",
+      type: "string",
+    },
+    {
+      name: "image",
+      title: "Image",
+      type: "url",
+    },
+    {
+      // this is only if you use credentials provider
+      name: "password",
+      type: "string",
+      hidden: true,
+    },
+  ],
+}
+```
+
+```ts
+// account
+
+export default {
+  name: "account",
+  title: "Account",
+  type: "document",
+  fields: [
+    {
+      name: "providerType",
+      type: "string",
+    },
+    {
+      name: "providerId",
+      type: "string",
+    },
+    {
+      name: "providerAccountId",
+      type: "string",
+    },
+    {
+      name: "refreshToken",
+      type: "string",
+    },
+    {
+      name: "accessToken",
+      type: "string",
+    },
+    {
+      name: "accessTokenExpires",
+      type: "string",
+    },
+    {
+      name: "user",
+      title: "User",
+      type: "reference",
+      to: { type: "user" },
+    },
+  ],
+}
+```
+
+### Sign Up and Sign In With Credentials
+
+### Setup
+
+`API Route`
+
+```ts
+// pages/api/sanity/signUp.ts
+
+import { signUpHandler } from "next-auth-sanity"
+import { NextApiRequest, NextApiResponse } from "next"
+import { client } from "your/sanity/client"
+
+export default (req: NextApiRequest, res: NextApiResponse) =>
+  signUpHandler({ req, res, client })
+```
+
+`Client`
+
+```ts
+import { signUp } from "next-auth-sanity/client"
+import { signIn } from "next-auth/client"
+
+const user = await signUp({
+  email,
+  password,
+  name,
+})
+
+await signIn("credentials", {
```

Comment:
```suggestion
await signIn("sanity-credentials", {
```

Reply:
the code is just a container for the official credential provider, in the future i will try to create a really custom provider